languages/rust/exercises/concept/string-vs-str/.meta/example.rs

```diff
@@ -0,0 +1,35 @@
+pub struct CSVBuilder {
+    content: String,
+}
+
+impl CSVBuilder {
+    // Create a new builder
+    pub fn new() -> Self {
+        Self { content: String::new() }
+    }
+
+    /// Adds an item to the list separated by a space and a comma.
+    pub fn add(&mut self, val: &str) {
+        if !self.content.is_empty() {
+            self.content.push(',');
+        }
+
+        if val.contains(",") || val.contains("\"") || val.contains("\n") {
+            self.content.push('"');
+            self.content.push_str(&val.replace("\"", "\"\""));
+            self.content.push('"');
+        } else {
+            self.content.push_str(val);
+        }
+    }
+
+    /// Start a new line
+    pub fn new_line(&mut self) {
+        self.content.push('\n');
+    }
```

Comment:
Nit: Suggest removing this function and `test_two_line`. It doesn't teach anything new, and makes it accurate to call this a `RecordBuilder`.

Reply:
A `Record` is a bit of a nebulous term and probably means different things to different people. 
Most deevelopers would know what CSV is and how to construct it (if they're coming from other languages) and if not there's a spec in the RFC that makes it clear. For a lot of people CSV is the familiar part that they don't need to learn.

Reply:
`CSVRecordBuilder`, then? 

> A `Record` is a bit of a nebulous term.

Note the term's usage in [para 2.1 of RFC 4180](https://tools.ietf.org/html/rfc4180#page-2):

> Each record is located on a separate line, delimited by a line break (CRLF).

I feel justified in using it precisely according to the spec.

In any case, my main objection to the term `CSVBuilder` is just that you'd pretty much never want to actually build a full CSV file like this; even if you wanted to build the records manually instead of `#[derive(Serialize)]`, you'd stream them out to some `W: Write` as each was completed. If we're not going to do it for real, and it doesn't provide any education not present elsewhere in the file, why bother making the students implement the ability to handle multiple lines? And if the builder can't write multiple lines, it's not a general-purpose CSV builder, but a record builder.

Reply:
CSVRecordBuilder I can get behind. `writeln!(self.content, "");` would do the trick cross platform. I do accept that doing a cross platform line ending is hard in rust. I asked recently on the internals and they said basically that `\r` was dead and that even notepad understood just `\n` these days... 

Maybe we just implement that function for them to show how its done.

Reply:
Ah your right let's keep it simple - I'll just say in the instructions that we're focusing on a single line.